src/nbl/video/CVulkanLogicalDevice.cpp

```diff
@@ -1164,7 +1168,15 @@ void CVulkanLogicalDevice::createComputePipelines_impl(
     for (const auto& info : createInfos)
     {
         initPipelineCreateInfo(outCreateInfo,info);
-        outCreateInfo->stage = getVkShaderStageCreateInfoFrom(info.shader,outShaderModule,outEntryPoints, outRequiredSubgroupSize,outSpecInfo,outSpecMapEntry,outSpecData);
+        const auto& spec = info.shader;
+        const auto debloatEntryPoints = std::array{ asset::ISPIRVDebloater::EntryPoint{spec.entryPoint , spec.stage} };
+        const auto debloatResult = m_spirvDebloater->debloat(spec.shader->getContent(), debloatEntryPoints, m_logger);
+        if (!debloatResult.isAllEntryPointsFound)
+        {
+            m_logger.log("Entry point %s is not found!", system::ILogger::ELL_ERROR, spec.entryPoint);
+            return;
+        }                 
```

Comment:
entry point removal and capability trimming need to be done before we hit `create...Pipelines_impl` 
https://github.com/Devsh-Graphics-Programming/Nabla/pull/839/files/7c548b7c80ff07d588a49571c7566b857ae9b215..31724ef75dc4e25499d95d318cd1024bce97b914#r2026648971